### PR TITLE
Add missing exports to json output

### PR DESCRIPTION
**Pull request by mosuem**
## Description
Add missing exports to JSON output if the `output` option is set for the `extract` command.

## Type of Change

- [x] 🚀 New feature (non-breaking change)
- [ ] 🛠️ Bug fix (non-breaking change)
- [x] ⚠️ Breaking change (feature or bug fix which breaks existing behaviors/APIs)
- [ ] 🏗️ Code refactor
- [ ] ⚙️ Build configuration change
- [ ] 📝 Documentation
- [ ] 🧹 Chore / Housekeeping

### CODE REVIEW

**Comment by mosuem** [comment]
I am unsure how to test this in your infrastructure - AFAIU, the tests for storage use `PackageApi` which cannot contains missing exports.

**Comment by devmil** [comment]
Hi @mosuem 
Wow, that was quick :) 
I will have a look as soon as possible.
Might take a few days, though.

**Comment by devmil** [comment]
> I am unsure how to test this in your infrastructure - AFAIU, the tests for storage use `PackageApi` which cannot contains missing exports.

I added an integration test.
`PackageApi` does contain all public API elements, including ones with no entry points attached to them, or did I miss your point?

**Comment by mosuem** [comment]
I thought the `PackageApi` contains only exported symbols. But an integration test works as well :)

**Comment by devmil** [comment]
> I thought the `PackageApi` contains only exported symbols. But an integration test works as well :)

Currently it does as it treats all used types implicitly as part of the public API. There is an open issue ( #170 ) that aims to remove types that are only passed in which would cause issues with the approach used here so it is important to either find another mechanism to get the list of unexported types once this issue gets solved (including the ones that then won't be part of the public API any more) or use the already existing `typeUsages` (https://github.com/bmw-tech/dart_apitool/blob/95277c3498cd77a2f19c9b094495b229e8555677/lib/src/model/interface_declaration.dart#L40) of the API model to filter out types that are only passed in when doing the diff (which I currently prefer)

I will update the issue accordingly.

File: test/package_api_storage_test.dart [hunk 1]
```diff
@@ -7,8 +7,9 @@ import 'package:test/test.dart';
 void main() {
   group('PackageApi storage tests', () {
     test('Storage v3 gets exported correctly', () {
-      final typedJson =
-          PackageApiStorage.packageApitoStorageJson(testPackage2Api);
+      final typedJson = PackageApiStorage.packageApitoStorageJson(
+        testPackage2Api,
```
Comment (adamgic) [inline test/package_api_storage_test.dart:11]: NIT: Not sure about the naming convention here, but after removing other tested objects we could probably remove the number, or start numbering from 1.
```suggestion
        testPackageApi,
```

**Review by adamgic — APPROVED** [review]

**Pull request merged**